docs/source/notes/extending.rst

```diff
@@ -72,6 +73,12 @@ Take the following steps:
   tensors if there are multiple outputs. Also, please refer to the
   docs of :class:`Function` to find descriptions of useful methods that can be
   called only from :meth:`~Function.forward`.
+- ``setup_context`` (optional). One can either write a "combined" :meth:`~Function.forward` that
+  accepts a ``ctx`` object or (as of PyTorch 2.0) a separate :meth:`~Function.forward` that does
+  not accept ``ctx`` and a ``setup_context`` method where the ``ctx`` modification happens.
+  In general the separate :meth:`~Function.forward` and ``setup_context`` is closer to how
+  PyTorch native operations work and therefore more composable with various PyTorch subsystems.
+  See :ref:`combining-forward-context` for more details
```

Comment:
We mention that we shouldn't do compute inside setup_context in an example somewhere above, but maybe we should mention that here too.